SafeExamBrowser.SystemComponents/VirtualMachineDetector.cs

```diff
@@ -33,6 +37,17 @@ public bool IsVirtualMachine()
 			isVirtualMachine |= manufacturer.Contains("parallels software");
 			isVirtualMachine |= model.Contains("virtualbox");
 			isVirtualMachine |= manufacturer.Contains("qemu");
+			
+			var macAddr = (from nic in NetworkInterface.GetAllNetworkInterfaces() where nic.OperationalStatus == OperationalStatus.Up select nic.GetPhysicalAddress().ToString()).FirstOrDefault();
+			isVirtualMachine |= ((byte.Parse(macAddr[1].ToString(), NumberStyles.HexNumber) & 2) == 2 || macAddr.StartsWith("080027"));
```

Comment:
Couldn't this code lead to an IndexOutOfRangeException or even a NullReferenceException, e.g. if there are no network interfaces?

Reply:
I'll be honest with you, I didn't think of a case like that even though it would be weird knowing the purpose of this program. I can change it so that it is only checked in case there is an interface.

Reply:
Yes, that would be great.